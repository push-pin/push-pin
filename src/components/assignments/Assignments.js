import React from 'react';
import PropTypes from 'prop-types';
import Assignment from './Assignment';

function Assignments({ assignments }) {
  const assignmentListItems = assignments.map(ass => {
    return (
      <li key={`${ass._id}-${ass._date}`}>
        <Assignment assignment={ass} />
      </li>
    );
  });

  return (
    <ul>
      {assignmentListItems}
    </ul>
  );
}

Assignments.propTypes = {
  assignments: PropTypes.array.isRequired
};

export default Assignments;
