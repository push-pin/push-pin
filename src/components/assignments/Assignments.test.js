import React from 'react';
import { shallow } from 'enzyme';
import Assignments from './Assignments';

describe('Assignments component', () => {
  it('renders a Assignments', () => {
    const assignments = [
      {
        classDate: '2012-10-15T21:26:17Z',
        dateDue: '2012-10-15T21:26:17Z',
        title:  'Ass 1',
        type: 'mob',
        submitted: false,
        _id: '1234'
      },
      {
        classDate: '2012-10-15T21:26:17Z',
        dateDue: '2012-10-15T21:26:17Z',
        title:  'Ass 11',
        type: 'mob',
        submitted: true,
        _id: '1234'
      },
      {
        classDate: '2012-10-15T21:26:17Z',
        dateDue: '2012-10-15T21:26:17Z',
        title:  'Ass 12',
        type: 'solo',
        submitted: false,
        _id: '3456'
      },
      {
        classDate: '2012-10-15T21:26:17Z',
        dateDue: '2012-10-15T21:26:17Z',
        title:  'Ass 13',
        type: 'reading',
        submitted: true,
        _id: '456'
      }
    ];
    
    const wrapper = shallow(<Assignments assignments={assignments} />);
    expect(wrapper).toMatchSnapshot();
  });
});
