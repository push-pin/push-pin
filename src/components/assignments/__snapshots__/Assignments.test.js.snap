// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignments component renders a Assignments 1`] = `
<ul>
  <Styled(styled.li)
    key="1234-undefined"
  >
    <DayItem
      _id="1234"
      classDate="2012-10-15T21:26:17Z"
      dateDue="2012-10-15T21:26:17Z"
      submitted={false}
      title="Ass 1"
      type="mob"
    />
  </Styled(styled.li)>
  <Styled(styled.li)
    key="1234-undefined"
  >
    <DayItem
      _id="1234"
      classDate="2012-10-15T21:26:17Z"
      dateDue="2012-10-15T21:26:17Z"
      submitted={true}
      title="Ass 11"
      type="mob"
    />
  </Styled(styled.li)>
  <Styled(styled.li)
    key="3456-undefined"
  >
    <DayItem
      _id="3456"
      classDate="2012-10-15T21:26:17Z"
      dateDue="2012-10-15T21:26:17Z"
      submitted={false}
      title="Ass 12"
      type="solo"
    />
  </Styled(styled.li)>
  <Styled(styled.li)
    key="456-undefined"
  >
    <DayItem
      _id="456"
      classDate="2012-10-15T21:26:17Z"
      dateDue="2012-10-15T21:26:17Z"
      submitted={true}
      title="Ass 13"
      type="reading"
    />
  </Styled(styled.li)>
</ul>
`;
